WordPress/gutenberg
# form-token-field/TokenInput: Fix automatic focus on first render due to updating selected value from async stores

## What?


Fixes #42285

Adds code to `TokenInput` to detect two things.
1. If it is the first render. This is determined by looking at the `value` variable. If one is passed in, this means there will be a dynamic selection index update which will cause focus.
2. Once the `TokenInput` gains focus by a user, first render goes to false.

## Why?


This is a nasty A11Y bug that needs to go away.

## How?


I think, magic.

## Testing Instructions




1. Have at least 25 users on a site with editor privileges or above.
2. Edit an existing post or page.
3. Notice how focus is not placed on the author combo box.

## Screenshots or screencast
